apps/cli/src/cli.ts

```diff
@@ -111,10 +116,17 @@ export async function cli(): Promise<void> {
 
   await logger.logHeader();
 
-  let goal: string | undefined = process.argv[2];
+  program
+    .option("-t, --timeout <number>")
+    .option("-g , --goal <string>")
+    .parse()
+
+  const options = program.opts()
+
+  let goal = options.goal ?? await logger.prompt("Enter your goal: ");
 
-  if (!goal) {
-    goal = await logger.prompt("Enter your goal: ");
+  if (options.timeout) {
```

Comment:
You can add this to the constructor instead (since it's not really meant to be updated while running)

Reply:
now we're passing `seconds` and `callback` of a timeout object to the constructor of Evo